modules/rosetta/src/main/scala/org/tessellation/rosetta/domain/Amount.scala

```diff
@@ -1,31 +1,59 @@
 package org.tessellation.rosetta.domain
 
+import cats.implicits.catsSyntaxEitherId
+
 import org.tessellation.ext.derevo.magnoliaCustomizable.snakeCaseConfiguration
-import org.tessellation.rosetta.domain.currency.Currency
+import org.tessellation.rosetta.domain.currency.{Currency, DAG}
+import org.tessellation.schema.transaction.{TransactionAmount, TransactionFee}
 
-import derevo.cats.eqv
+import derevo.cats.{eqv, show}
 import derevo.circe.magnolia._
 import derevo.derive
 import eu.timepit.refined.api.Refined
 import eu.timepit.refined.auto._
 import eu.timepit.refined.boolean.Not
 import eu.timepit.refined.cats._
 import eu.timepit.refined.generic.Equal
+import eu.timepit.refined.numeric.{NonNegative, Positive}
+import eu.timepit.refined.refineV
 import io.circe.refined._
 import io.estatico.newtype.macros.newtype
 
 object amount {
-  @derive(eqv, customizableDecoder, customizableEncoder)
+  type AmountValuePredicate = Not[Equal[0L]]
+  type AmountValueRefined = Long Refined AmountValuePredicate
+
+  @derive(eqv, decoder, encoder, show)
+  @newtype
+  case class AmountValue(value: AmountValueRefined) {
+    def isNegative: Boolean = value < 0L
+    def isPositive: Boolean = value > 0L
+
+    def negate: Either[String, AmountValue] =
+      if (value.value == Long.MinValue)
+        "Long.MinValue cannot be negated".asLeft
```

Comment:
Can it be a case class extending NoStackTrace or Throwable? Exactly as we do in the rest of the codebase?

Reply:
@marcinwadon @TheMMaciek 
Given the discussion so far doesn't that mean that the refined type for `AmountValue` -- `Not[Equal[0L]]` -- is not correct? The context of how we're using `Amount` requires every amount have a negation, so having an amount of `Long.MinValue` should not be allowed.

Reply:
That's a fair point. The max value (or positive value) is always negateable, whereas the min long value is not negateable but at the same time it's not a value that can ever occur. So you might be right that the `AmountValue` should be refined more to not allow the long min value as it's not correct in the current domain.